File: repository/tasks_repository.py
import os
from utils.task_status_enum import TaskStatus
import mysql.connector

db_host = os.getenv("DBHost")
db_username = os.getenv("DBUsername")
db_password = os.getenv("DBPassword")

def connect_database():
    return mysql.connector.connect(
        user=db_username, 
        password=db_password, 
        host=db_host, 
        port=3306
    )

def get_task_id_by_filepath(db_connection, filepath):
    cursor = db_connection.cursor()
    cursor.execute("SELECT id from imagefunctions.tasks WHERE filepath = '%s'" % (filepath))
    return cursor.fetchone()[0]

def update_task_status(db_connection, status: TaskStatus, task_id):
    cursor = db_connection.cursor()
    cursor.execute("UPDATE imagefunctions.tasks SET Status='%s' WHERE Id = '%s'" % (status.value, task_id))
    db_connection.commit()

File: api-rest/repository/tasksRepository.js
const mysql = require('mysql2/promise');

const hostName = process.env.HOST_NAME || 'hostName';
const userName = process.env.USER_NAME || 'userName';
const password = process.env.PASSWORD || 'password';

async function connect(){
    const connection = await mysql.createConnection({
        host: hostName,
        user: userName,
        password: password,
        database: 'imagefunctions'
    });
    console.log("Connected to MySQL");
    return connection;
}

async function findById(id) {
    const conn = await connect();

    const [rows] = await conn.execute(
        'SELECT * FROM imagefunctions.tasks WHERE id = ?',
        [id]
    );

    await conn.close();
    console.log("Connection closed");

    return await rows[0];
}

async function save(filepath) {
    const conn = await connect();
    const now = new Date();
    const [{ insertId }] = await conn.execute(
        'INSERT INTO imagefunctions.tasks (filepath, status, createdDate, lastModifiedDate) VALUES (?, ?, ?, ?)', [filepath, 'PENDING', now, now]
    );
    await conn.close();

    return await insertId;
}

exports.findById = findById;
exports.save = save;
